mw_jwt.go

```diff
@@ -206,22 +225,41 @@ func (k *JWTMiddleware) getBasePolicyID(token *jwt.Token) (string, bool) {
 	return "", false
 }
 
+func (k *JWTMiddleware) getUserIdFromClaim(token *jwt.Token) (string, error) {
+	var userId string
+	var found bool
+
+	if k.Spec.JWTIdentityBaseField != "" {
+		userId, found = token.Claims.(jwt.MapClaims)[k.Spec.JWTIdentityBaseField].(string)
+
+		if !found {
+			log.Warning("Base Field not found, falling back to SUB")
+		}
+	}
+	if !found {
+		userId, found = token.Claims.(jwt.MapClaims)[SUB].(string)
+		if !found {
+			message := fmt.Sprint("user id was not found in claims")
+			log.Error(message)
+			return "", errors.New(message)
+		}
+	}
+	log.Debugf("Found User Id: ", userId)
```

Comment:
I'd still follow `logrus`' convention to do `log.WithField("userId", userId).Debug("Found User Id")` as its API was done specifically to remove this formatted outputs.